### ISSUE TITLE
Issues upgrading stock Moodle to IOMAD

### ISSUE THREAD

**Issue by davidpesce** [iomad/iomad#1200]
IOMAD 3.6 (latest)

When attempting to add a user as a department manager an error is thrown.

```
Coding error detected, it must be fixed by a programmer: Invalid call to role_unassign(), please use role_unassign_all() when removing multiple role assignments
```

```
×Stack trace:
line 1594 of /lib/accesslib.php: coding_exception thrown
line 1136 of /local/iomad/lib/company.php: call to role_unassign()
line 166 of /blocks/iomad_company_admin/company_managers_form.php: call to company::upsert_company_user()
line 322 of /blocks/iomad_company_admin/company_managers_form.php: call to company_managers_form->process()
```

```
×Output buffer: <br /> <b>Notice</b>: Trying to get property of non-object in <b>/local/iomad/lib/company.php</b> on line <b>1136</b><br />
```

**Comment by davidpesce** [comment]
OK, figured this one out. It happened on two upgraded instances. One from Moodle 3.3 to IOMAD 3.6 and the other from Moodle 3.4 to IOMAD 3.6. 

All of the IOMAD roles were not created during the upgrade process. I do have a test IOMAD 3.6 instance I built from scratch that had the proper roles. I exported/imported them and it all works fine now. 

If I have some more time, I'll come back and try to determine where the bug exists and submit a PR.
